features/messages/impl/src/main/kotlin/io/element/android/features/messages/impl/timeline/factories/TimelineItemsFactory.kt

```diff
@@ -87,7 +87,16 @@ class TimelineItemsFactory @Inject constructor(
                     newTimelineItemStates.add(timelineItemState)
                 }
             } else {
-                newTimelineItemStates.add(cacheItem)
+                val updatedItem = if (cacheItem is TimelineItem.Event && roomMembers.isNotEmpty()) {
+                    eventItemFactory.update(
+                        timelineItem = cacheItem,
+                        receivedMatrixTimelineItem = timelineItems[index] as MatrixTimelineItem.Event,
```

Comment:
Are you sure that `timelineItems[index]` can always be cast to `MatrixTimelineItem.Event`?

Reply:
It uses the unique id to identify cached items, so if the cached item is an event I expect the updated item with that id to also be an event. Maybe that's not always the case?

Reply:
Yes the cached value is obviously the same, otherwise there is a big issue :D